.agents/skills/advantagescope/scripts/capture_layout.py

```
@@ -0,0 +1,156 @@
+    while time.time() - start < 30 and (port is None or token is None):
+        line = process.stdout.readline() if process.stdout is not None else ""
+        if line:
+            print(line, end="")
+            port_match = re.search(r"127\.0\.0\.1:(\d+)", line)
+            if port_match:
+                port = int(port_match.group(1))
+            token_match = re.search(r"AdvantageScope agent token: (\S+)", line)
+            if token_match:
+                token = token_match.group(1)
+        elif process.poll() is not None:
+            raise SystemExit(f"AdvantageScope exited early with code {process.returncode}")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)